File: Day6/Reverse Integer/7.cpp
#include <cmath>

class Solution {
public:
    int reverse(int x) {
        int ans = 0;
        int count = 0;
        
        //These were declared becuase VSCode could not find INT_MAX and INT_MIN
        int INT_MAX = pow(2, 31) - 1;
        int INT_MIN = pow(-2, 31);

        while(x != 0)
        {
            int digit = x % 10;
            
            if(ans > INT_MAX/10 || ans < INT_MIN/10)
            {
                return 0;
            }
            
            ans = ans * 10 + digit;
            
            count ++;
            x /= 10;
        }
        
        return ans;
    }
};

File: Day6/Reverse Integer/7.py
import sys

class Solution:
    def reverse(self, x: int) -> int:
        ans = 0
        isNegative = False
        
        if x < 0:
            isNegative = True
            x *= -1
        
        while x != 0:
            digit = x % 10
        
            if (ans > (2**31 - 1) // 10) or (ans < (-2**31) // 10):
                return 0
        
            ans = ans * 10 + digit
            
            x //= 10
        
        if isNegative:
            ans *= -1
            
        return ans
        